fairlearn/metrics/_metric_frame.py

```diff
@@ -493,7 +580,14 @@ def group_max(self, errors: str = "raise") -> Union[Any, pd.Series, pd.DataFrame
             The maximum value over sensitive features. The exact type
             follows the table in :attr:`.MetricFrame.overall`.
         """
-        return self.__group("max", errors)
+        if errors not in _VALID_ERROR_STRING:
+            raise ValueError(_INVALID_ERRORS_VALUE_ERROR_MESSAGE)
+
+        value = self._result_cache["group_max"][errors]
+        if isinstance(value, Exception):
+            raise value
```

Comment:
what does this look like? It's odd to be raising something somewhere else without `from` maybe? And should we have like a `_raise_or_return(value)` method instead?

Reply:
yeah I got the logic, what I was wondering was, imagine there's a non-fairlearn exception happening there, and we raise it here. How odes that look to the end user, and would they be able to figure out what to do, since the stack trace is now not very obvious to them? Could we make up an example where we could see how it looks like maybe?